### PR TITLE
mk - update unit conversion to more generic

### PR DESCRIPTION
**Pull request by 7sne**

### CODE REVIEW

**Review by hasparus — APPROVED** [review]
LGTM.

File: pages/eth-unit-conversion.tsx [hunk 1]
```diff
@@ -22,8 +23,13 @@ export default function EthUnitConversion() {
   function handleChangeEvent(value: string, unitType: UnitType) {
     let out
 
+    // 'On paste' conversion from hexadecimal to decimal values
+    if (value.split('')[1] === 'x') {
```
Comment (krzkaczor) [inline pages/eth-unit-conversion.tsx:27]: Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith instead

File: pages/eth-unit-conversion.tsx [hunk 2]
```diff
@@ -22,8 +23,13 @@ export default function EthUnitConversion() {
...
+      value = parseInt(value, 16).toString()
```
Comment (krzkaczor) [inline pages/eth-unit-conversion.tsx:28]: please move this piece to a separate function and add tests

File: lib/convertUnits.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { BigNumber } from 'bignumber.js'
```
Comment (krzkaczor) [inline lib/convertUnits.ts:1]: I'd say that this file has a problem with public interface not being clearly visible. 

Everything is exported but only some of the things are used (or tested) by external modules.

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: lib/convertUnits.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  toUnit: T,
+  precisionDict: PrecisionDict<T>,
+): string | undefined {
+  return new BigNumber(value)
+    .times(unitPrecisionToValue(precisionDict[fromUnit]))
+    .div(unitPrecisionToValue(precisionDict[toUnit]))
+    .toString()
+}
+
+export function convertEthUnits(value: string, fromUnit: UnitType, toUnit: UnitType): string | undefined {
```
Comment (krzkaczor) [inline lib/convertUnits.ts:19]: I'd move `convertEthUnits` and `convertTokenUnits` to the beginning of the file as these functions are public interface of the module. @hasparus thoughts?

**Pull request merged**